src/approaches/nsrt_learning_approach.py

```diff
@@ -53,12 +54,90 @@ def learn_from_offline_dataset(self, dataset: Dataset) -> None:
+            if os.path.exists(dataset_fname):
+                # Load the ground atoms dataset.
+                with open(dataset_fname, "rb") as f:
+                    ground_atom_dataset_atoms = pkl.load(f)
+                assert len(trajectories) == len(ground_atom_dataset_atoms)
+                logging.info("\n\nLOADED GROUND ATOM DATASET")
+
+                if CFG.env == "behavior":  # pragma: no cover
+                    pred_name_to_pred = {
+                        pred.name: pred
+                        for pred in self._get_current_predicates()
+                    }
+                    new_ground_atom_dataset_atoms = []
+                    # Since we save ground atoms for behavior with dummy
+                    # classifiers, we need to restore the correct classifers.
+                    for ground_atom_seq in ground_atom_dataset_atoms:
+                        new_ground_atom_seq = []
+                        for ground_atom_set in ground_atom_seq:
+                            new_ground_atom_set = {
+                                GroundAtom(
+                                    pred_name_to_pred[atom.predicate.name],
+                                    atom.entities)
+                                for atom in ground_atom_set
+                            }
+                            new_ground_atom_seq.append(new_ground_atom_set)
+                        new_ground_atom_dataset_atoms.append(
+                            new_ground_atom_seq)
+
+                # The saved ground atom dataset consists only of sequences
+                # of sets of GroundAtoms, we need to recombine this with
+                # the LowLevelTrajectories to create a GroundAtomTrajectory.
+                ground_atom_dataset = []
+                for i, traj in enumerate(trajectories):
+                    ground_atom_seq = ground_atom_dataset_atoms[i]
+                    ground_atom_dataset.append(
+                        (traj, [set(atoms) for atoms in ground_atom_seq]))
+            else:
+                raise ValueError(f"Cannot load ground atoms: {dataset_fname}")
+        else:
+            # Apply predicates to data, producing a dataset of abstract states.
+            ground_atom_dataset = utils.create_ground_atom_dataset(
+                trajectories, self._get_current_predicates())
+            # Save ground atoms dataset to file. Note that a
+            # GroundAtomTrajectory contains a normal LowLevelTrajectory and a
+            # list of sets of GroundAtoms, so we only save the list of
+            # GroundAtoms (the LowLevelTrajectories are saved separately).
+            ground_atom_dataset_to_pkl = []
+            for gt_traj in ground_atom_dataset:
+                trajectory = []
+                for ground_atom_set in gt_traj[1]:
+                    if CFG.env == "behavior":  # pragma: no cover
+                        # In the case of behavior, we cannot directly pickle
```

Comment:
if we're going to go this far, we might as well just always strip away the classifiers when saving atoms to disk? that seems reasonable to me given that in general the classifiers could be complicated non-pickleable things (and in fact they are, for behavior). but Tom might disagree on this (his preference is higher prio)